node/src/components/block_proposer.rs

```diff
@@ -281,13 +259,15 @@ impl BlockProposer {
         &mut self,
```

Comment:
Not necessarily in this PR, but:
* We should probably rename this method… `remaining_deploys` sounds like some simple getter.
* Collecting all the `past_deploys` below in each call doesn't scale. Instead of storing them by block, should we store all finalized deploys in a single big collection?

Reply:
Sounds reasonable to me :+1: